core/src/main/java/org/kiwix/kiwixmobile/core/CoreApp.kt

```diff
@@ -74,11 +74,12 @@ abstract class CoreApp : Application() {
   override fun onCreate() {
     super.onCreate()
     instance = this
+    fileLogger = FileLogger()
     coreComponent = DaggerCoreComponent.builder()
       .context(this)
       .build()
     AndroidThreeTen.init(this)
-    writeLogFile()
+    fileLogger.writeLogFile(this)
```

Comment:
and move this line below the inject, calling `inject(this)` is what tells to dagger to find fields annotated with `@Inject` inside `this` and call their constructors or retrieve an instance